File: Codeforces Problems/candies.c
#include <stdio.h>
#include <math.h>
int main()
{
    int t;
    scanf("%d", &t);

    while (t--)
    {
        int n, x = 1;
        scanf("%d", &n);

        long int sum = 0;

        while (n % (sum + (int)pow(2, x)) == 0)
        {
            sum += pow(2, x);
            x++;
        }

        printf("%d\n", x);
    }
    return 0;
}

File: Codeforces Problems/Candies.java
import java.util.*;

public class Candies {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        while (t-- > 0) {
            int n = sc.nextInt(), i = 3, j = 1;

            while (n % i != 0) {
                j++;
                i += Math.pow(2, j);
            }

            System.out.println(n / i);
        }
    }
}